src/lib/__tests__/InstantSearch-test.js

```diff
@@ -90,6 +90,61 @@ See documentation: https://www.algolia.com/doc/api-reference/widgets/instantsear
 `);
   });
 
+  it('throws a warning when insights is detected but insightsClient was not passed', () => {
+    warning.cache = {};
+
+    const AlgoliaAnalyticsObject = 'aa';
+    global.AlgoliaAnalyticsObject = AlgoliaAnalyticsObject;
+    global[AlgoliaAnalyticsObject] = jest.fn();
+
+    expect(() => {
+      // eslint-disable-next-line no-new
+      new InstantSearch({
+        indexName: 'indexName',
+        searchClient: createSearchClient(),
+      });
+    })
+      .toWarnDev(`[InstantSearch.js]: InstantSearch detected the Insights client in the global scope.
+To connect InstantSearch to the Insights client, make sure to specify the \`insightsClient\` option:
+
+const search = instantsearch({
+  /* ... */
+  insightsClient: window.aa,
+});
+
+See documentation: https://www.algolia.com/doc/api-reference/widgets/instantsearch/js/`);
+
+    delete global.AlgoliaAnalyticsObject;
+    delete global[AlgoliaAnalyticsObject];
+  });
+
+  it('does not throw a warning when insights is detected but insightsClient was passed', () => {
+    warning.cache = {};
+
+    const AlgoliaAnalyticsObject = 'aa';
+    global.AlgoliaAnalyticsObject = AlgoliaAnalyticsObject;
+    global[AlgoliaAnalyticsObject] = jest.fn();
+
+    expect(() => {
+      // eslint-disable-next-line no-new
+      new InstantSearch({
+        indexName: 'indexName',
+        searchClient: createSearchClient(),
+        insightsClient: jest.fn(),
+      });
+    }).not
+      .toWarnDev(`[InstantSearch.js]: InstantSearch detected the insights client in the global scope. 
```

Comment:
This assertion is outdated now because the message expectation differs from the real one. We could use `.not.toWarnDev()` here to avoid that.

Reply:
The message has been updated here: https://github.com/algolia/instantsearch.js/pull/4191/commits/64b4133917f57d5bde9e964bd0af5e1cf1388fc1

Reply:
For this test _not_ to send a false positive in the future, we should either store the message in a variable to use it in the two tests or simply use `.not.toWarnDev()`.